tools/pipelines-tasks/AVDAppAttachPublish/main.ts

```diff
@@ -0,0 +1,98 @@
+import { ToolRunner } from 'azure-pipelines-task-lib/toolrunner';
+import tl = require('azure-pipelines-task-lib/task');
+import path = require('path');
+import armStorage = require('azure-pipelines-tasks-azure-arm-rest/azure-arm-storage');
+import { AzureRMEndpoint } from 'azure-pipelines-tasks-azure-arm-rest/azure-arm-endpoint';
+import { AzureEndpoint, StorageAccount } from 'azure-pipelines-tasks-azure-arm-rest/azureModels';
+import * as os from 'os';
+
+const HELPER_SCRIPT = path.join(__dirname, 'publishAVD.ps1');
+const TARGET_DLL = path.join(__dirname, '/AppAttachFrameworkDLL/AppAttachKernel.dll');
+
+function isNonEmpty(str: string): boolean {
+    return (!!str && !!str.trim());
+}
+
+function getResourceGroupNameFromUri(resourceUri: string): string {
+    if (isNonEmpty(resourceUri)) {
+        resourceUri = resourceUri.toLowerCase();
+        return resourceUri.substring(resourceUri.indexOf("resourcegroups/") + "resourcegroups/".length, resourceUri.indexOf("/providers"));
```

Comment:
Nit: If you're handling URIs I would rather you used the built in URI functionality instead of doing string manipulation to be safe

Reply:
Updated.